From f687f45111f586c986b46b14dc502e3078e46d86 Mon Sep 17 00:00:00 2001
From: Bruno Martins <bgcngm@gmail.com>
Date: Mon, 9 Jun 2025 17:57:59 +0100
Subject: [PATCH] power: Support MODE_EXT and SET_INTERACTIVE_EXT as external
 modules

In certain cases, providing the source files is not possible at all,
so allow linking against libs.

Change-Id: I0dda703d2ed98e2b2da224c1fa3c1978fbf2f38c
---
 Android.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Android.mk b/Android.mk
index c4dc8d9..80eeb65 100644
--- a/Android.mk
+++ b/Android.mk
@@ -64,11 +64,17 @@ endif # End of board specific list
 ifneq ($(TARGET_POWERHAL_MODE_EXT),)
     LOCAL_CFLAGS += -DMODE_EXT
     LOCAL_SRC_FILES += ../../../../$(TARGET_POWERHAL_MODE_EXT)
+else ifneq ($(TARGET_POWERHAL_MODE_EXT_LIB),)
+    LOCAL_CFLAGS += -DMODE_EXT
+    LOCAL_STATIC_LIBRARIES += $(TARGET_POWERHAL_MODE_EXT_LIB)
 endif
 
 ifneq ($(TARGET_POWERHAL_SET_INTERACTIVE_EXT),)
     LOCAL_CFLAGS += -DSET_INTERACTIVE_EXT
     LOCAL_SRC_FILES += ../../../../$(TARGET_POWERHAL_SET_INTERACTIVE_EXT)
+else ifneq ($(TARGET_POWERHAL_SET_INTERACTIVE_EXT_LIB),)
+    LOCAL_CFLAGS += -DSET_INTERACTIVE_EXT
+    LOCAL_STATIC_LIBRARIES += $(TARGET_POWERHAL_SET_INTERACTIVE_EXT_LIB)
 endif
 
 ifneq ($(TARGET_TAP_TO_WAKE_NODE),)
